test/e2e/singlecluster/statefulset_test.go

```diff
@@ -372,5 +380,50 @@ var _ = ginkgo.Describe("StatefulSet integration", func() {
 				gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
 			})
 		})
+
+		ginkgo.It("should delete all Pods if StatefulSet was deleted after being partially ready", func() {
```

Comment:
Actually, how is this test related to the issue? Is the new assert "Check all pods are deleted" not enough to repro the issue?

Reply:
To reliably catch this error, we can check if the StatefulSet was deleted after being partially ready. However, it can also happens when we just delete StatefulSet.

Reply:
> So the new check "Check all pods are deleted" does not catch the issue reliably?

Yes, exactly.

>  I'm not seeing a difference between partially ready StatefulSet and fully ready. Do you know why it matters for the change of catching the issue?

To be honest, I'm not sure why. But on this way I'm catching an error everytime.

Reply:
I see, I'm ok to keep it then